codersdk/deployment.go

```diff
@@ -2595,22 +2597,17 @@ Write out the current server config as YAML to stdout.`,
 			YAML:        "thresholdDatabase",
 			Annotations: serpent.Annotations{}.Mark(annotationFormatDuration, "true"),
 		},
-		// Email options
-		emailFrom,
-		emailSmarthost,
-		emailHello,
-		emailForceTLS,
-		emailAuthIdentity,
-		emailAuthUsername,
-		emailAuthPassword,
-		emailAuthPasswordFile,
-		emailTLSStartTLS,
-		emailTLSServerName,
-		emailTLSSkipCertVerify,
-		emailTLSCertAuthorityFile,
-		emailTLSCertFile,
-		emailTLSCertKeyFile,
 		// Notifications Options
+		{
+			Name:        "Notifications: Enabled",
+			Description: "Controls if notifications are enabled.",
+			Flag:        "notifications-enabled",
+			Env:         "CODER_NOTIFICATIONS_ENABLED",
```

Comment:
Can you please add some tests validating this functionality?
For example, we need to ensure enqueuing notifications doesn't cause issues when notifications are disabled, etc (although I don't see anything in this PR relating to that - which is a smell).

Reply:
Sure thing!

This PR does make a small change to the logic of setting the enqueuer

The enqueuer is by default set as the `notifications.NoopEnqueuer`.

https://github.com/coder/coder/blob/c51311310575c9be8894367b4bc0b8e91e764034/cli/server.go#L624

We then re-assign the enqueuer only if notifications are enabled.

https://github.com/coder/coder/blob/c51311310575c9be8894367b4bc0b8e91e764034/cli/server.go#L905-L914

Reply:
Ah, you're right! Great 👍